### PR TITLE
WAT-885

### PR DESCRIPTION
**Pull request by ByMsx**

### CODE REVIEW

File: packages/plugin-selenium-driver/src/plugin/index.ts [hunk 1]
```diff
@@ -493,7 +468,7 @@ export class SeleniumPlugin implements IBrowserProxyPlugin {
         const client = this.getBrowserClient(applicant);
```
Comment (flops) [inline packages/plugin-selenium-driver/src/plugin/index.ts]: Rename this into `selector` and make it into 
`const selector = await this.getBrowserClient(applicant).$(xpath)`.

File: packages/plugin-selenium-driver/src/types.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import { Config } from 'webdriverio';
 
-export type SeleniumPluginConfig = Config & {
+export type SeleniumPluginConfig = WebdriverIO.RemoteOptions & Config & {
     chromeDriverPath?: string;
     recorderExtension: boolean;
     clientCheckInterval: number;
     clientTimeout: number;
+    host?: string; // fallback for configuration. In WebdriverIO 5 field host renamed to hostname
+    desiredCapabilities?: any; // fallback for configuration. In WebdriverIO 5 field renamed
```
Comment (flops) [inline packages/plugin-selenium-driver/src/types.ts]: Is there real structure for capabilities object?

File: packages/plugin-selenium-driver/src/plugin/index.ts [hunk 3]
```diff
@@ -421,358 +429,301 @@ export class SeleniumPlugin implements IBrowserProxyPlugin {
...
-        }
+        const selector = await client.$(xpath);
+        return selector.waitForExist(timeout);
     }
 
     public async waitForVisible(applicant: string, xpath: string, timeout: number) {
         await this.createClient(applicant);
         const client = this.getBrowserClient(applicant);
 
         if (client) {
```
Comment (flops) [inline packages/plugin-selenium-driver/src/plugin/index.ts]: if not needed

File: packages/plugin-selenium-driver/src/plugin/index.ts [hunk 4]
```diff
@@ -421,358 +429,301 @@ export class SeleniumPlugin implements IBrowserProxyPlugin {
...
-        if (client) {
-            const elements = await client.findElements('xpath', xpath) as unknown;
-            const a = (elements as Array<Record<string, string>>).map((o) => {
-                const keys = Object.keys(o);
-                return { 'ELEMENT': o[keys[0]] };
-            });
-
-            return a;
-        }
+        const elements = await client.findElements('xpath', xpath) as unknown;
```
Comment (flops) [inline packages/plugin-selenium-driver/src/plugin/index.ts]: Is it needed to cast `uknowkn` and then back to `Array<Record<string, string>>` why not to do it right in place?

File: packages/plugin-selenium-driver/src/plugin/index.ts [hunk 5]
```diff
@@ -421,358 +429,301 @@ export class SeleniumPlugin implements IBrowserProxyPlugin {
...
     }
 
     public async keysOnElement(applicant: string, xpath: string, value: any) {
         await this.createClient(applicant);
         const client = this.getBrowserClient(applicant);
 
-        if (client) {
-            return (client as any).keysOnElement(xpath, value);
-        }
+        return (client as any).keysOnElement(xpath, value);
```
Comment (flops) [inline packages/plugin-selenium-driver/src/plugin/index.ts]: `any` still needed?

Comment (ByMsx) [reply packages/plugin-selenium-driver/src/plugin/index.ts]: Yes. But we can describe our interface of BrowserObject.

**Pull request merged**